modules/cluster/src/main/java/io/fluo/cluster/runner/YarnAppRunner.java

```diff
@@ -84,64 +88,98 @@ private synchronized TwillRunnerService getTwillRunner() {
         throw new IllegalStateException(e);
       }
     }
-    return twillRunner;
+    return twillRunners.get(config.getApplicationName());
   }
 
-  private void checkIfInitialized() {
+  private void checkIfInitialized(FluoConfiguration config) {
+
     try (FluoAdminImpl admin = new FluoAdminImpl(config)) {
       if (!admin.zookeeperInitialized()) {
-        System.out.println("ERROR - A Fluo '" + config.getApplicationName()
-            + "' application has not been initialized yet in Zookeeper at "
-            + config.getAppZookeepers());
-        System.exit(-1);
+        throw new FluoException("A Fluo '" + config.getApplicationName() + "' application has not "
+            + "been initialized yet in Zookeeper at " + config.getAppZookeepers());
       }
     }
   }
 
-  @Override
-  public void start() {
-    checkIfInitialized();
+  public void list(FluoConfiguration config) {
+    try (CuratorFramework curator = CuratorUtil.newFluoCurator(config)) {
+      curator.start();
 
-    if (twillIdExists()) {
-      String runId = getTwillId();
+      try {
+        List<String> children = curator.getChildren().forPath("/");
+        if (children.isEmpty()) {
+          System.out.println("Fluo instance (" + config.getInstanceZookeepers() + ") does not "
+              + " contain any applications");
+        } else {
+          System.out.println("Fluo instance (" + config.getInstanceZookeepers() + ") contains "
+              + children.size() + " applications\n");
+          System.out.println("Application         Status");
+          System.out.println("-----------         ------");
+          for (String path : children) {
+            System.out.format("%-20s", path);
+            FluoConfiguration appConfig = new FluoConfiguration(config);
+            appConfig.setApplicationName(path);
+            if (!twillIdExists(appConfig)) {
+              System.out.println("STOPPED");
```

Comment:
Yes. UNKNOWN is now printed if there is no Twill ID found.  This could occur if the application has not been started yet or the path was deleted in ZK.  Fixed in 60aed4c1f